arangod/RocksDBEngine/RocksDBRestReplicationHandler.cpp

```diff
@@ -888,13 +888,24 @@ void RocksDBRestReplicationHandler::handleCommandRevisionTree() {
   // smaller and thus improve efficiency)
   bool onlyPopulated = _request->parsedValue("onlyPopulated", false);
 
-  auto tree = ctx.collection->getPhysical()->revisionTree(ctx.batchId);
+  std::string collectionName = _request->value("collection");
+
+  std::unique_ptr<containers::RevisionTree> tree;
 
   {
     RocksDBReplicationContext* c = _manager->find(ctx.batchId);
     RocksDBReplicationContextGuard guard(_manager, c);
+    if (c != nullptr) {  // we have the RocksDBReplicationContext!
+      // See if we can get the revision tree from the context:
+      tree = c->getPrefetchedRevisionTree(collectionName);
+      // This might return a nullptr!
+    }
+    if (tree == nullptr) {  // Still not there, try to get it directly:
+      tree = ctx.collection->getPhysical()->revisionTree(
+          c->snapshot()->GetSequenceNumber());
+    }
     if (c != nullptr) {
-      c->removeBlocker(_request->databaseName(), _request->value("collection"));
+      c->removeBlocker(_request->databaseName(), collectionName);
```

Comment:
It is to me what the exact contents of `collectionName` should be here. Is it a shard name or a collection guid?
It would be good to document the expected contents here, so that one does not mix shard names and guids by accident.

Reply:
I have renamed the variables in all places to say "guid" and documented this.